lib/image_composition.dart

```diff
@@ -0,0 +1,167 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'extensions.dart';
+
+export 'extensions.dart';
+
+class _Composed {
+  /// The image that will be composed.
+  final Image image;
+
+  /// The position where the [image] will be composed.
+  final Vector2 position;
+
+  /// The source on the [image] that will be composed.
+  final Rect source;
+
+  final double rotation;
+
+  final Vector2 anchor;
+
+  final bool isAntiAlias;
+
+  /// The [BlendMode] that will be used when composing the [image].
+  final BlendMode blendMode;
+
+  _Composed(
+    this.image,
+    this.position,
+    this.source,
+    this.rotation,
+    this.anchor,
+    this.isAntiAlias,
+    this.blendMode,
+  );
+}
+
+class ImageComposition {
+  /// The values that will be used to compose the image
+  final List<_Composed> _composes = [];
+
+  /// The [defaultBlendMode] can be used to change how each image will be
+  /// blended onto the composition. Defaults to [BlendMode.srcOver].
+  final BlendMode defaultBlendMode;
+
+  /// The [defaultAntiAlias] can be used to if each image will be anti aliased.
+  final bool defaultAntiAlias;
+
+  ImageComposition({
+    this.defaultBlendMode = BlendMode.srcOver,
+    this.defaultAntiAlias = false,
+  })  : assert(defaultBlendMode != null, 'defaultBlendMode can not be null'),
+        assert(defaultAntiAlias != null, 'defaultAntiAlias can not be null');
+
+  /// Add an image to the [ImageComposition].
+  ///
+  /// The [image] will be added at the given [position] on the composition.
+  ///
+  /// An optional [source] can be used to only add the data that is within the
+  /// [source] of the [image].
+  ///
+  /// An optional [rotation] (in radians) can be used to rotate the image when it
```

Comment:
would be good to document from which axis this rotates and in what direction (cw or ccw)